### PR TITLE
Simple changes to support polling of both SQS and instance metadata

### LINKED ISSUE #4: Add support for spot instances watching for marked-for-termination state
Shudder could also hit instance metadata and call a webhook when it detects that a spot instance is going down. As spot instance termination wouldn't trigger ordinary lifecycle actions, this might be useful.

http://aws.amazon.com/blogs/aws/new-ec2-spot-instance-termination-notices/

### PR DESCRIPTION
**Pull request by avram**
Should resolve #4.

### CODE REVIEW

File: shudder/metadata.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+"""Module to set up polling of instance metadata for the termination of a spot instance
+
+"""
+import requests
+
+
+def poll_instance_metadata():
+    """Check instance metadata for a scheduled termination"""
+    r = requests.get("http://169.254.169.254/latest/meta-data/spot/termination-time")
+    return r.status_code < 400
```
Comment (avram) [inline shudder/metadata.py]: It appears that the endpoint 404's when there is no termination time set.

Comment (Raynes) [inline shudder/metadata.py]: lol less than 400; classy

File: shudder/queue.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 
 
 def create_queue():
-    """Creates the sqs queue and returns the connection/queue"""
+    """Creates the SQS queue and returns the connection/queue"""
```
Comment (Raynes) [inline shudder/queue.py]: OH COME ON

File: shudder/queue.py [hunk 3]
```diff
@@ -56,14 +56,17 @@ def should_terminate(msg):
...
+    sns_conn.unsubscribe(sns_arn)
+
+
+def poll_queue(conn, queue):
+    """Poll SQS until we get a termination message."""
+    message = queue.read()
+    if message:
+        conn.delete_message(queue, message)
+        if should_terminate(message):
+            return True
```
Comment (Raynes) [inline shudder/queue.py]: Isn't this just `should_terminate(message)`? If True then True else False? :P

Comment (avram) [inline shudder/queue.py]: Indeed. :)

**Pull request merged**